beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go

```diff
@@ -68,11 +68,14 @@ func (vs *Server) getExecutionPayload(ctx context.Context, slot types.Slot, vIdx
 		copy(pid[:], payloadId[:])
 		payloadIDCacheHit.Inc()
 		payload, err := vs.ExecutionEngineCaller.GetPayload(ctx, pid)
-		if err != nil {
-			return nil, err
+		switch {
+		case err == nil:
+			warnIfFeeRecipientDiffers(payload, feeRecipient)
+			return payload, nil
+		case errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded):
```

Comment:
on `context.Canceled`  we should abort anyway,  this is passed by the calling function isn't it?